### ISSUE TITLE
picture thumbnails are wrong in the "Auto upload" and "Upload" pages of app

### ISSUE THREAD

**Issue by S74HK9hV** [nextcloud/android#2177]
### Actual behaviour
- photo thumbnails in the Auto upload page are wrong. showing old photos that i deleted a long time ago. In the upload window, the thumbnail for the photo being auto-uploaded is also wrong.

### Expected behaviour
- Thumbnails should show the correct photos that actually exist on the device. 
- thumbnails should show the correct photo being uploaded to server
 
### Steps to reproduce
1. take some photos
2. delete one or more of those photos
3. wait until auto-upload triggers
4. check thumbnails in the "Auto upload" and "Upload" pages

### Environment data
Android version: 8.0.0

Device model: Sony Xperia X Compact

Stock or customized system: stock

Nextcloud app version: 3.0.1

Nextcloud server version: 13

#### Web server error log
```
no errors server side
```

#### Nextcloud log (data/nextcloud.log)
```
no errors server side
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
I cannot reproduce this. 
For local files we are using hash code as identifier for cached thumbnails. The hash code should be unique.

**Comment by prefect4742** [comment]
I reproduced this on app v3.0.2 with this procedure:

1. take some photos
2. (unsure) wait for them to upload
3. delete the latest photos
4. take some new photos (which will get the same file names as the deleted ones on devices that follow DCF naming)
5. wait for new photos to upload

I got both "local file not found" with old thumbnails as well as multiple uploads of the same filename as before but with the old thumbnail.
The reason I'm unsure about step 2 is that I'm not sure how the app reacts when it gets wifi but can't reach the server... how long will it wait until it retries? My server is not reachable from the internet so uploads only work in my home wifi (and even that is not 100% due to some unrelated DHCP issues...)
